pulp_container/app/registry_api.py

```diff
@@ -951,13 +993,30 @@ def handle_safe_method(self, request, path, pk):
             try:
                 tag = models.Tag.objects.get(name=pk, pk__in=repository_version.content)
             except models.Tag.DoesNotExist:
-                raise ManifestNotFound(reference=pk)
+                if distribution.remote:
```

Comment:
here we need to add a check if this distro is part of pull-through-distro otherwise we enable pull-through caching for regular synced repos.